### PR TITLE
Split output filetype variable into atm and sfc

### LINKED ISSUE #600: Improve organization of FV3 output filetype settings
**Description**
The determination of the FV3 model output filetype is a bit unwieldy because it is combining two filetype (atm and sfc) that need determined separately based on resolution and machine performance.

**Acceptance Criteria (Definition of Done)**
Separate output filetype variables for atm and sfc that are set with other resolution-dependent settings and later combined to form the appropriate namelist entry.

**Dependencies**
None

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
The determination of the FV3 output filetype was a bit unwieldy as it
contained two different settings (one for the atmosphere and one for
the surface) that were not necessarily changed at the same time. They
were also being determined in a different location than other settings
based on the model resolution.

Now the old OUTPUT_FILETYPES variable has been split into two different
variables, OUTPUT_FILETYPES_ATM and OUTPUT_FILETYPES_SFC. The determi-
nation was also moved into the config.fv3 file, where other resolution-
dependent computational settings are set.

This has resulted in some functional change on WCOSS Dell, as the EnKF
was using different chunking settings and never used parallel output
for the surface likely because wasn't being run at a low enough resolu-
tion to consider it. However, there is no reason to believe the two
forecast modes should have different chunk or output settings. So, I've
used the EnKF chunking settings and the free forecast switchover point
for the surface output mode. However, on Hera the EnKF still forces 
serial output, as there is some issue with parallel netcdf and ensembles 
on that machine.

The new filetype settings are now also divorced from the OUTPUT_FILE
setting. However, that setting will soon be unnecessary as nemsio
is removed from the code as an option (see Issue #601).

Fixes: #600
Refs: #601

**Type of change**
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**How Has This Been Tested?**
- [x] Cycled test on Hera C384/C196
- [x] Cylced test on Orion C384/C196
- [x] Cycled test on WCOSS Dell C768/C384
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Leaving as draft until I repeat the test on WCOSS Dell.

Adding GDAS team as reviewers since this technically effects them.

**Comment by WalterKolczynski-NOAA** [comment]
Actually, I think I need to run this again at a higher resolution since these are so resolution dependent. I only ran the same C384/C196 test I've run previously. I don't anticipate any code changes though.

File: ush/parsing_model_configure_FV3.sh
Comment (junwang-noaa) [inline ush/parsing_model_configure_FV3.sh:41]: These two settings are resolution dependent, "parallel_netcdf" is not recommended to lower resolution than C384L127 as it takes longer. For C384L127, the "parallel_netcdf" runs faster for atmf files, but not for sfcf file for C768L127, it is recommended to used "parallel_netcdf" for both atmf and sfcf files. If the workflow will also be used by low resolution runs, especially long forecast run, then using "netcdf" will be a better choice.

File: parm/config/config.fv3
Comment (junwang-noaa) [inline parm/config/config.fv3]: May I ask what is the reason to choose these specific chunksizes? What benefit do we get?

Comment (aerorahul) [reply parm/config/config.fv3]: They are just being moved from `config.fcst` to `config.fv3`

**Comment by junwang-noaa** [comment]
I remember I had a discussion with Kate. The ichunk2d should be same as the
output domain IMO/JMO, which is equal to the (4*RESTILE) and (2*RESTILE)
at this time, but it would be good to set inchunk2d/ichunk3d as "IMO"
directly., same for JMO

File: ush/parsing_model_configure_FV3.sh
Comment (aerorahul) [reply ush/parsing_model_configure_FV3.sh:41]: @junwang-noaa 
We are choosing to defining these variables based on the resolution in `config.fv3`. 
There is a `default` also set at the top there.

File: parm/config/config.fv3
Comment (WalterKolczynski-NOAA) [reply parm/config/config.fv3]: All chunk settings were pulled from what was already being used in the EnKF or free forecast previously.

**Comment by WalterKolczynski-NOAA** [comment]
> I remember I had a discussion with Kate. The ichunk2d should be same as the output domain IMO/JMO, which is equal to the (4*RESTILE) and (2*RESTILE) at this time, but it would be good to set inchunk2d/ichunk3d as "IMO" directly., same for JMO

If this is true for all resolutions, we can just pull all that out of the CASE block and set it in one place.

**Comment by WalterKolczynski-NOAA** [comment]
System now runs on all three core machines.

**Review by aerorahul — APPROVED** [review]
Looks good. 
Approve pending testing.

**Comment by WalterKolczynski-NOAA** [comment]
> Looks good. Approve pending testing.

Testing is already complete. Still hoping someone from DA will sign off since there is a difference in chunking and serial/parallel at some resolutions.

**Comment by WalterKolczynski-NOAA** [comment]
@RussTreadon-NOAA @CatherineThomas-NOAA can one of you give this a quick look before I merge?

**Comment by CoryMartin-NOAA** [comment]
Were any timing tests performed with new vs old chunk sizes for both forecast and GSI? I seem to remember a great deal of variability in both write-grid component write speed and GSI read speed based on the choices selected for chunk size.

See: https://docs.google.com/spreadsheets/d/1DPXJ6-3OXzn67-7ZK_VRttDl1kwYPhf-FBndtg9ZAnM/edit?usp=sharing from @RussTreadon-NOAA

**Comment by WalterKolczynski-NOAA** [comment]
> Were any timing tests performed with new vs old chunk sizes for both forecast and GSI? I seem to remember a great deal of variability in both write-grid component write speed and GSI read speed based on the choices selected for chunk size.
> 
> See: https://docs.google.com/spreadsheets/d/1DPXJ6-3OXzn67-7ZK_VRttDl1kwYPhf-FBndtg9ZAnM/edit?usp=sharing from @RussTreadon-NOAA

I didn't do any timing tests, but the 3d chunking should be unchanged at operational resolutions. This is mostly extending the same chunking rubric to coarser resolutions and using the 2d chunk rubric EnKF was already doing in other forecast jobs.

**Review by CatherineThomas-NOAA — COMMENTED** [review]
The changes look okay to me, though I'm not knowledgeable about chunking and will defer to someone who knows more from the DA side (@CoryMartin-NOAA? @jswhit2?)

**Pull request merged**
